### PR TITLE
feat: Surveys + opening urls support

### PR DESCRIPTION
**Pull request by aymanenadi**
https://github.com/lifeomic/react-native-sdk/assets/19395435/872c656c-2690-4aa3-abba-ea5fdac19061

### CODE REVIEW

File: src/navigators/TabNavigator.tsx [hunk 1]
```diff
@@ -31,6 +31,7 @@ export function TabNavigator() {
       <Tab.Screen
         name="HomeTab"
         component={HomeStack}
+        key="HomeTab"
```
Comment (aymanenadi) [inline src/navigators/TabNavigator.tsx:34]: This is unrelated. I kept getting warnings about missing keys.

File: src/screens/AuthedAppTileScreen.tsx [hunk 2]
```diff
@@ -60,8 +69,36 @@ export const AuthedAppTileScreen = ({
...
+    oauthCallbackUrl,
+    activeSubjectId,
+    searchParams,
+  ]);
+
+  const handlePageLoaded = () => {
+    setIsPageLoaded(true);
+  };
+
+  const dispatchLXFocusEvent = useCallback(() => {
```
Comment (sternetj) [inline src/screens/AuthedAppTileScreen.tsx:91]: It would be nice at some point if we could change `lx` to something more meaningful.

File: src/hooks/useHandleAppTileEvents.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      case AppTileMessageType.openUrl:
+        handleOpenUrlMessage(appTileMessage);
+        break;
+
+      // no default
+    }
+  };
+
+  const handleAppTileNavigationStateChange = (event: WebViewNavigation) => {
+    if (event.url.indexOf('pending-lx-action') !== -1) {
```
Comment (sternetj) [inline src/hooks/useHandleAppTileEvents.ts:100]: It would be nice at some point if we could change `lx` to something more meaningful.

**Review by sternetj — APPROVED** [review]

**Review by evanheisler — APPROVED** [review]
This is pretty much all ported over from the applet, correct?

File: src/hooks/useHandleAppTileEvents.ts (hunk 3 above)
Comment (aymanenadi) [reply src/hooks/useHandleAppTileEvents.ts:100]: I agree. I didn't change it in this PR because of the tight deadlines. We'll need to update all the app tiles that use this route, while still keeping compatibility with the LIFE apps / PMA.

**Comment by aymanenadi** [comment]
> This is pretty much all ported over from the applet, correct?

Do you mean LIFE Extend? For the most part, yes. The main difference is how we create a new code every time a new app tile is opened since the codes are for one-time use only.

**Comment by evanheisler** [comment]
> > This is pretty much all ported over from the applet, correct?
> 
> Do you mean LIFE Extend? For the most part, yes. The main difference is how we create a new code every time a new app tile is opened since the codes are for one-time use only.

Ah, yes thank you. That answers my question then — I was curious what was new code vs. ported over.

File: src/hooks/useHandleAppTileEvents.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const { routeName, params = {} } = appletMessage.data;
+    if (routeName === 'tiles/Today/Survey') {
+      openSurveyAppTile(params.questionnaire as string | undefined);
+    } else {
+      console.warn('Unsupported route name. Ignoring...', { routeName });
+    }
+  };
+
+  const enforceWebURLProtocol = (url: string) => {
+    const allowedProtocols = ['https://', 'http://'];
```
Comment (markdlv) [inline src/hooks/useHandleAppTileEvents.ts:72]: Any reason you know of why we need to support `http://`.  If you're not sure, it'd be great to start this initial migration with `https://` being the only option.

Comment (markdlv) [reply src/hooks/useHandleAppTileEvents.ts:72]: Oh, I see - this is used by `handleOpenUrlMessage`, which I'm betting is for program actions that like a youtube URL, article to review, or journal/info website?  Got it if so - if we allow program configuration with either scheme, probably best to keep both allowed here as well.  It's forcibly non-authed URLs here, which is key.

File: src/hooks/useHandleAppTileEvents.ts (hunk 3 above)
Comment (markdlv) [reply src/hooks/useHandleAppTileEvents.ts:100]: +1 on both fronts

File: src/hooks/index.ts [hunk 5]
```diff
@@ -19,3 +19,4 @@ export * from './useSubjectProjects';
 export * from './useTheme';
 export * from './useUser';
 export * from './useWearables';
+export * from './useHandleAppTileEvents';
```
Comment (markdlv) [inline src/hooks/index.ts:22]: Sorry for the nit but can you alphabetize this?

**Review by markdlv — APPROVED** [review]

File: src/hooks/index.ts (hunk 5 above)
Comment (aymanenadi) [reply src/hooks/index.ts:22]: Of course. I'll do it in a follow-up PR.

File: src/hooks/useHandleAppTileEvents.ts (hunk 4 above)
Comment (aymanenadi) [reply src/hooks/useHandleAppTileEvents.ts:72]: That's correct. This is for URLs entered by the admin in the program builder pointing to articles to read, etc. We do not currently enforce a particular protocol.
